src/ts/uimanager.ts

```diff
@@ -573,451 +465,12 @@ export class UIManager {
-            new PlaybackToggleButton(),
-            new AdMessageLabel(),
-            new VolumeControlButton(),
-            new FullscreenToggleButton(),
-          ],
-        }),
-        new AdSkipButton(),
-      ],
-      cssClasses: ['ui-skin-legacy', 'ui-skin-ads'],
-    });
-  }
-
-  function legacyCastReceiverUI() {
-    let controlBar = new ControlBar({
-      components: [
-        new SeekBar(),
-        new PlaybackTimeLabel(),
-      ],
-    });
-
-    return new UIContainer({
-      components: [
-        new SubtitleOverlay(),
-        new PlaybackToggleOverlay(),
-        new Watermark(),
-        controlBar,
-        new TitleBar(),
-        new ErrorMessageOverlay(),
-      ],
-      cssClasses: ['ui-skin-legacy', 'ui-skin-cast-receiver'],
-    });
-  }
-
-  function legacyTestUI() {
-    let settingsPanel = new SettingsPanel({
-      components: [
-        new SettingsPanelItem('Video Quality', new VideoQualitySelectBox()),
-        new SettingsPanelItem('Audio Track', new AudioTrackSelectBox()),
-        new SettingsPanelItem('Audio Quality', new AudioQualitySelectBox()),
-        new SettingsPanelItem('Subtitles', new SubtitleSelectBox()),
-      ],
-      hidden: true,
-    });
-
-    let controlBar = new ControlBar({
-      components: [settingsPanel,
-        new PlaybackToggleButton(),
-        new SeekBar({ label: new SeekBarLabel() }),
-        new PlaybackTimeLabel(),
-        new VRToggleButton(),
-        new VolumeToggleButton(),
-        new VolumeSlider(),
-        new VolumeControlButton(),
-        new VolumeControlButton({ vertical: false }),
-        new SettingsToggleButton({ settingsPanel: settingsPanel }),
-        new CastToggleButton(),
-        new FullscreenToggleButton(),
-      ],
-    });
-
-    return new UIContainer({
-      components: [
-        new SubtitleOverlay(),
-        new CastStatusOverlay(),
-        new PlaybackToggleOverlay(),
-        new Watermark(),
-        new RecommendationOverlay(),
-        controlBar,
-        new TitleBar(),
-        new ErrorMessageOverlay(),
-      ],
-      cssClasses: ['ui-skin-legacy'],
-    });
-  }
-
-  export function buildLegacyUI(player: PlayerAPI, config: UIConfig = {}): UIManager {
-    return new UIManager(player, [{
-      ui: legacyAdsUI(),
-      condition: (context: UIConditionContext) => {
-        return context.isAd && context.adClientType === 'vast';
-      },
-    }, {
-      ui: legacyUI(),
-    }], config);
-  }
-
-  export function buildLegacyCastReceiverUI(player: PlayerAPI, config: UIConfig = {}): UIManager {
-    return new UIManager(player, legacyCastReceiverUI(), config);
-  }
-
-  export function buildLegacyTestUI(player: PlayerAPI, config: UIConfig = {}): UIManager {
-    return new UIManager(player, legacyTestUI(), config);
-  }
+/**
+ * Export the {@link UIFactory} as `UIManager.Factory` for backwards compatibility with the UI module in player v8 RCs.
+ * @deprecated remove with UI v4
+ */
+export namespace UIManager {
```

Comment:
The player v8 RCs do have their suitable UI version bundled. 
Do we really need this backwards compatibility for RCs since there was no stable release of the player or the UI?

Reply:
For now we need it, else it doesn't work with the player. We can adjust the UI module with the next player RC and then remove this, but since an RC must have a stable API I went with this approach. Or course one can argue that this is not related to the player API and therefore a valid change.